front/gatsby/src/ContextGate.tsx

```diff
@@ -23,9 +26,13 @@ export function ContextGate({
     <AlertsContextProvider>
       <AccountsContextProvider>
         <TxQueueContextProvider>
-          <ApiContextProvider provider={new WsProvider(ARCHIVE_NODE_ENDPOINT)}>
-            <StakingContextProvider>{children}</StakingContextProvider>
-          </ApiContextProvider>
+          <HealthContextProvider provider={wsProvider}>
+            <HealthGate>
+              <ApiContextProvider loading={<Loading active>Initializing chain...</Loading>} provider={wsProvider}>
+                {children}
```

Comment:
The only context I would put now is AccountsContextProvider. Maybe AlertsContextProvider if that fits sveta's designs.